src/Build/BackEnd/Components/Logging/LoggingService.cs

```diff
@@ -244,16 +244,23 @@ internal partial class LoggingService : ILoggingService, INodePacketHandler
         /// </summary>
         private ConcurrentQueue<object> _eventQueue;
 
+        // Long-lived event handles that never get disposed to avoid race conditions.
+        private readonly AutoResetEvent _longLivedDequeueEvent = new AutoResetEvent(false);
+        private readonly ManualResetEvent _longLivedEmptyQueueEvent = new ManualResetEvent(true);
+        private readonly AutoResetEvent _longLivedEnqueueEvent = new AutoResetEvent(false);
```

Comment:
Why are those needed?

Reply:
The idea was to keep long living events and just reset their state when StartLoggingEventProcessing() is called.